plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java

```diff
@@ -841,6 +875,46 @@ public static Response registerRemoteModel(String name, String connectorId) thro
             .makeRequest(client(), "POST", "/_plugins/_ml/models/_register", null, TestHelper.toHttpEntity(registerModelEntity), null);
     }
 
+    public static Response registerRemoteModelWithTTL(String name, String connectorId, int ttl) throws IOException {
+        String registerModelGroupEntity = "{\n"
+            + "  \"name\": \"remote_model_group\",\n"
+            + "  \"description\": \"This is an example description\"\n"
+            + "}";
+        Response response = TestHelper
+            .makeRequest(
+                client(),
+                "POST",
+                "/_plugins/_ml/model_groups/_register",
+                null,
+                TestHelper.toHttpEntity(registerModelGroupEntity),
+                null
+            );
+        Map responseMap = parseResponseToMap(response);
+        assertEquals((String) responseMap.get("status"), "CREATED");
+        String modelGroupId = (String) responseMap.get("model_group_id");
+
+        String registerModelEntity = "{\n"
+            + "  \"name\": \""
+            + name
+            + "\",\n"
+            + "  \"function_name\": \"remote\",\n"
+            + "  \"model_group_id\": \""
+            + modelGroupId
+            + "\",\n"
+            + "  \"version\": \"1.0.0\",\n"
+            + "  \"description\": \"test model\",\n"
+            + "  \"connector_id\": \""
+            + connectorId
+            + "\",\n"
+            + "  \"deploy_setting\": "
+            + " { \"model_ttl_minutes\": "
```

Comment:
`model_ttl_minutes` can this be float? Like if I want `0.5` minutes => 30 seconds?

Reply:
In addition, I'm curious to know more about this `model_ttl_minutes`. Let's say, I set `model_ttl_minutes` to 1 minutes and the model got deployed. Now at around 50th Second I made a prediction request with that model. In that case will the model still get undeployed after 10 seconds or it will start waiting for another 60 seconds?

Reply:
TTL resets when the model is used in Prediction. It's only based on idling time. The number of ttl is a long number because Java Duration doesn't support float number. If you need to use seconds, the time unit needs to change to seconds. But TTL Time granularity at minutes is fine enough.